### PR TITLE
Refactor MTurk html and server setup

### PR DESCRIPTION
**Pull request by yf225**
1. Setting up the AWS server no longer requires information about the HIT. Instead the HIT config would be loaded from a separate file called ``hit_config.json``.

2. We now support custom cover page and chat interface for MTurk agent, so that different types of MTurk agent can be shown a different UI.

### CODE REVIEW

File: parlai/mturk/core/agents.py [hunk 1]
```diff
@@ -58,7 +60,14 @@ def init_aws(self, opt):
             return
 
         print('Setting up MTurk backend...')
-        html_api_endpoint_url, json_api_endpoint_url, requester_key_gt = setup_aws(task_description=opt['task_description'], num_hits=opt['num_hits'], num_assignments=opt['num_assignments'], is_sandbox=opt['is_sandbox'])
+        create_hit_config(task_description=opt['task_description'], num_hits=opt['num_hits'], num_assignments=opt['num_assignments'], is_sandbox=opt['is_sandbox'])
+        if not self.task_files_to_copy:
```
Comment (harmdevries89) [inline parlai/mturk/core/agents.py]: @yf225 shouldn't you check here if each mturk_agent_id has its templates in task_files_to_copy?

File: parlai/mturk/core/setup_aws.py [hunk 2]
```diff
@@ -233,7 +243,24 @@ def setup_rds():
 
     return host
 
-def setup_relay_server_api(mturk_submit_url, rds_host, task_description, is_sandbox, num_hits, num_assignments, requester_key_gt, should_clean_up_after_upload=True):
+def create_hit_config(task_description, num_hits, num_assignments, is_sandbox):
```
Comment (harmdevries89) [inline parlai/mturk/core/setup_aws.py]: This looks much better

File: parlai/mturk/tasks/model_evaluator/run.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2017-present, Facebook, Inc.
+1# Copyright (c) 2017-present, Facebook, Inc.
```
Comment (alexholdenmiller) [inline parlai/mturk/tasks/model_evaluator/run.py]: oops

File: parlai/mturk/core/agents.py (hunk 1 above)
Comment (yf225) [inline parlai/mturk/core/agents.py]: @harmdevries89 I found that setup_aws.py:create_zip_file() would skip missing files, so I guess we don't have to check here

File: parlai/mturk/core/agents.py [hunk 4]
```diff
@@ -31,6 +31,7 @@
 polling_interval = 1 # in seconds
 create_hit_type_lock = threading.Lock()
 local_db_lock = threading.Lock()
+task_dir = os.getcwd()
```
Comment (yf225) [inline parlai/mturk/core/agents.py]: I agreed - os.getcwd() is not guaranteed to return the directory that the file lives in. Changing it to be using the method you suggested.

File: parlai/mturk/tasks/model_evaluator/run.py (hunk 3 above)
Comment (yf225) [inline parlai/mturk/tasks/model_evaluator/run.py]: oops my bad :/

**Comment by yf225** [comment]
Zhilin's issue seems to be on the number of concurrent database queries during HIT creation which overwhelms the RDS instance. Currently making it so that we are doing bulk message inserts (1 SQL statement for whatever number of new messages we send, and doing it every 1 second). Still need to test for stability and other potential issues and I am planning to run it tonight.

Also added script for automatically completing HITs by sending dummy text. This can replace manual testing for all of our tasks, and it would be especially useful when we want to test for scalability issues.

**Comment by yf225** [comment]
Some investigation results: currently we are using t2.micro for RDS instance, which can only support dozens of concurrent connections (i.e. there can only be dozens of Turkers who have the mturk page open at the same time). I am thinking of creating larger instances when the user runs a larger number of HITs, to avoid the database overload issue. I will create another pull request for it.

**Comment by harmdevries89** [comment]
@yf225 when do you plan to merge this PR? Zhilin's issue seems orthogonal to the refactoring, so I hope we can merge this one soon.

File: parlai/mturk/core/auto_test.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+'''
```
Comment (alexholdenmiller) [inline parlai/mturk/core/auto_test.py]: Nit: ''' => """

**Pull request merged**